core/lib/types/src/snapshots.rs

```diff
@@ -38,36 +42,131 @@ pub struct SnapshotStorageLogsChunkMetadata {
     pub filepath: String,
 }
 
-#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct SnapshotStorageLogsStorageKey {
     pub l1_batch_number: L1BatchNumber,
     pub chunk_id: u64,
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
+#[derive(Debug, Clone, PartialEq)]
 pub struct SnapshotStorageLogsChunk {
     pub storage_logs: Vec<SnapshotStorageLog>,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct SnapshotStorageLog {
     pub key: StorageKey,
     pub value: StorageValue,
     pub l1_batch_number_of_initial_write: L1BatchNumber,
     pub enumeration_index: u64,
 }
 
-#[derive(Debug, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
+#[derive(Debug, PartialEq)]
 pub struct SnapshotFactoryDependencies {
     pub factory_deps: Vec<SnapshotFactoryDependency>,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct SnapshotFactoryDependency {
     pub bytecode: Bytes,
 }
+
+impl ProtoFmt for SnapshotFactoryDependency {
+    type Proto = crate::proto::SnapshotFactoryDependency;
+    fn read(r: &Self::Proto) -> anyhow::Result<Self> {
+        Ok(Self {
+            bytecode: Bytes(required(&r.bytecode).context("bytecode")?.clone()),
+        })
+    }
+    fn build(&self) -> Self::Proto {
+        Self::Proto {
+            bytecode: Some(self.bytecode.0.as_slice().into()),
+        }
+    }
+}
+
+impl ProtoFmt for SnapshotFactoryDependencies {
+    type Proto = crate::proto::SnapshotFactoryDependencies;
+    fn read(r: &Self::Proto) -> anyhow::Result<Self> {
```

Comment:
Nit: Blank lines between items (like in `impl ProtoFmt for SnapshotStorageLog`) would help readability.

Reply:
corrected